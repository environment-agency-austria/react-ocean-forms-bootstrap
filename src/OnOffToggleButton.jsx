/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { ButtonGroup, Button, Input } from 'reactstrap';
import { fieldMetaShape, fieldShape, withForm } from 'react-ocean-forms';

import FieldLine from './FieldLine';

/**
 * Component for displaying bootstrap
 * form groups with an html input and
 * oForm support
 */
class OnOffToggleButton extends React.Component {
  constructor(props) {
    super(props);

    this.onRadioBtnClick = this.onRadioBtnClick.bind(this);
  }

  onRadioBtnClick(value) {
    const { field: { onChange, name } } = this.props;
    onChange({
      target: {
        name,
        value,
      },
    });
  }

  render() {
    const {
      field,
      onLabel,
      offLabel,
      meta: {
        plaintext,
        stringFormatter,
      },
      field: {
        value,
      },
    } = this.props;
    const onValue = stringFormatter(onLabel);
    const offValue = stringFormatter(offLabel);
    const isOn = !(value === '' || value === false);

    if (plaintext) {
      return (
        <FieldLine {...this.props}>
          <Input {...field} plaintext>
            {isOn ? onValue : offValue}
          </Input>
        </FieldLine>
      );
    }

    return (
      <FieldLine {...this.props}>
        <ButtonGroup id={field.id}>
          <Button
            id={`${field.id}-on`}
            color="primary"
            onClick={() => this.onRadioBtnClick(true)}
            active={isOn}
          >
            {onValue}
          </Button>
          <Button
            id={`${field.id}-off`}
            color="primary"
            onClick={() => this.onRadioBtnClick(false)}
            active={!isOn}
          >
            {offValue}
          </Button>
        </ButtonGroup>
      </FieldLine>
    );
  }
}

OnOffToggleButton.displayName = 'OnOffToggleButton';

OnOffToggleButton.defaultProps = {
  onLabel: 'ojs_togglebutton_on',
  offLabel: 'ojs_togglebutton_off',
};

OnOffToggleButton.propTypes = {
  label: PropTypes.string.isRequired,
  onLabel: PropTypes.string,
  offLabel: PropTypes.string,
  meta: fieldMetaShape.isRequired,
  field: fieldShape.isRequired,
};

export const BaseOnOffToggleButton = OnOffToggleButton;
export default withForm(OnOffToggleButton);
