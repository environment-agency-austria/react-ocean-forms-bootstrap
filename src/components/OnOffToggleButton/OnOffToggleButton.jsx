/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { ButtonGroup, Button, Input } from 'reactstrap';
import {
  FormText,
  fieldMetaShape,
  fieldShape,
  withForm,
} from 'react-ocean-forms';

import FieldLine from '../FieldLine';

/**
 * Component for displaying bootstrap
 * form groups with an html input and
 * oForm support
 */
class OnOffToggleButton extends React.Component {
  constructor(props) {
    super(props);

    this.onRadioBtnClick = this.onRadioBtnClick.bind(this);
  }

  onRadioBtnClick(value) {
    const { field: { onChange, name } } = this.props;
    onChange({
      target: {
        name,
        value,
      },
    });
  }

  render() {
    const {
      field,
      onLabel,
      offLabel,
      meta: {
        plaintext,
      },
      field: {
        value,
      },
    } = this.props;

    const isOn = !(value === '' || value === false);

    if (plaintext) {
      return (
        <FieldLine {...this.props}>
          <Input {...field} plaintext>
            <FormText text={isOn ? onLabel : offLabel} />
          </Input>
        </FieldLine>
      );
    }

    return (
      <FieldLine {...this.props}>
        <ButtonGroup id={field.id}>
          <Button
            id={`${field.id}-on`}
            color="primary"
            onClick={() => this.onRadioBtnClick(true)}
            outline
            active={isOn}
            disabled={field.disabled}
          >
            <FormText text={onLabel} />
          </Button>
          <Button
            id={`${field.id}-off`}
            color="primary"
            onClick={() => this.onRadioBtnClick(false)}
            outline
            active={!isOn}
            disabled={field.disabled}
          >
            <FormText text={offLabel} />
          </Button>
        </ButtonGroup>
      </FieldLine>
    );
  }
}

OnOffToggleButton.displayName = 'OnOffToggleButton';

OnOffToggleButton.defaultProps = {
  onLabel: 'ojs_togglebutton_on',
  offLabel: 'ojs_togglebutton_off',
};

OnOffToggleButton.propTypes = {
  label: PropTypes.string.isRequired,
  onLabel: PropTypes.string,
  offLabel: PropTypes.string,
  meta: fieldMetaShape.isRequired,
  field: fieldShape.isRequired,
};

export const BaseOnOffToggleButton = OnOffToggleButton;
export default withForm(OnOffToggleButton);
