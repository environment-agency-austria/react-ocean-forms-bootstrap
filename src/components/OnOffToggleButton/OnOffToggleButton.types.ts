import { IUseFieldProps } from 'react-ocean-forms';

import { IBaseFieldLineProps } from '../FieldLine';

/**
 * Props for the OnOffToggleButton component
 */
export interface IOnOffToggleButtonProps<TSubmitValue> extends IUseFieldProps<boolean, TSubmitValue>, IBaseFieldLineProps {
  /**
   * Label for the on-button
   */
  onLabel?: string;
  /**
   * Label for the off-button
   */
  offLabel?: string;
}
