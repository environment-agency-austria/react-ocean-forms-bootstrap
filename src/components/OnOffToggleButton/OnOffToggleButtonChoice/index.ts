export * from './OnOffToggleButtonChoice';
export * from './OnOffToggleButtonChoice.types';
