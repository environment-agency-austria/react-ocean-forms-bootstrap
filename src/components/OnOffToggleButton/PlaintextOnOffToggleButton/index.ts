export * from './PlaintextOnOffToggleButton';
export * from './PlaintextOnOffToggleButton.types';
