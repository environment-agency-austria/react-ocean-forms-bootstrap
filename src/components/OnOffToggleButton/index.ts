export * from './OnOffToggleButton';
export * from './OnOffToggleButton.types';
