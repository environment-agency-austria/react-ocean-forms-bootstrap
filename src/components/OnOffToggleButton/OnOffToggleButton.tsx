/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { useCallback } from 'react';

import { useField } from 'react-ocean-forms';
import {  ButtonGroup } from 'reactstrap';

import { FieldLine } from '../FieldLine';
import { IOnOffToggleButtonProps } from './OnOffToggleButton.types';
import { PlaintextOnOffToggleButton } from './PlaintextOnOffToggleButton';
import { OnOffToggleButtonChoice } from './OnOffToggleButtonChoice';

/**
 * Component for displaying bootstrap
 * form groups with an html input and
 * oForm support
 */
export const OnOffToggleButton = <TSubmitValue extends unknown = boolean>(props: IOnOffToggleButtonProps<TSubmitValue>): JSX.Element => {
  const {
    onLabel = 'ojs_togglebutton_on',
    offLabel = 'ojs_togglebutton_off',
    ...rest
  } = props;

  const { fieldProps, metaProps } = useField(rest);

  const { onChange } = fieldProps;
  const onRadioBtnClick = useCallback((value: boolean) => {
    onChange({
      target: {
        value,
      }
    });
  }, [onChange]);

  if (metaProps.plaintext) {
    return (
      <PlaintextOnOffToggleButton
        fieldProps={fieldProps}
        metaProps={metaProps}
        onLabel={onLabel}
        offLabel={offLabel}
        {...rest}
      />
    );
  }

  return (
    <FieldLine {...rest} fieldProps={fieldProps} metaProps={metaProps}>
      <ButtonGroup id={fieldProps.id}>
        <OnOffToggleButtonChoice
          variant="on"
          label={onLabel}
          fieldProps={fieldProps}
          onClick={onRadioBtnClick}
        />
        <OnOffToggleButtonChoice
          variant="off"
          label={offLabel}
          fieldProps={fieldProps}
          onClick={onRadioBtnClick}
        />
      </ButtonGroup>
    </FieldLine>
  );
};
