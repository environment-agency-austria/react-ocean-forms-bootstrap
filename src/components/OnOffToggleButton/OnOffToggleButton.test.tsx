import React from 'react';

import { render, fireEvent } from '@testing-library/react';
import { Form } from 'react-ocean-forms';
import { OnOffToggleButton } from './OnOffToggleButton';

describe('<OnOffToggleButton />', () => {
  it('should render correctly', () => {
    const { asFragment, getByText, getByLabelText } = render(
      <Form>
        <OnOffToggleButton name="mock" label="mock" />
      </Form>
    );

    expect(getByLabelText('mock')).toBeVisible();

    expect(getByText('ojs_togglebutton_on')).toBeVisible();
    expect(getByText('ojs_togglebutton_on')).toHaveClass('active');

    expect(getByText('ojs_togglebutton_off')).toBeVisible();
    expect(getByText('ojs_togglebutton_off')).not.toHaveClass('active');

    expect(asFragment()).toMatchSnapshot();
  });

  describe('plaintext mode', () => {
    it('should render plaintext Button', () => {
      const { asFragment, getByDisplayValue, queryByDisplayValue, getByLabelText } = render(
        <Form plaintext>
          <OnOffToggleButton name="mock" label="mock" />
        </Form>
      );

      expect(getByLabelText('mock')).toBeVisible();

      expect(getByDisplayValue('ojs_togglebutton_on')).toBeVisible();
      expect(getByDisplayValue('ojs_togglebutton_on')).toHaveAttribute('readonly');

      expect(queryByDisplayValue('ojs_togglebutton_off')).toBeNull();

      expect(asFragment()).toMatchSnapshot();
    });

    it('should render the correct label in plaintext mode', () => {
      const { getByDisplayValue, queryByDisplayValue, getByLabelText } = render(
        <Form plaintext>
          <OnOffToggleButton name="mock" label="mock" onLabel="mock on label" offLabel="mock off label" value={false} />
        </Form>
      );

      expect(getByLabelText('mock')).toBeVisible();

      expect(getByDisplayValue('mock off label')).toBeVisible();
      expect(getByDisplayValue('mock off label')).toHaveAttribute('readonly');

      expect(queryByDisplayValue('mock on label')).toBeNull();
    });
  });

  describe('disabled mode', () => {
    it('should render correctly in disabled mode', () => {
      const { getByText } = render(
        <Form disabled>
          <OnOffToggleButton name="mock" label="mock" />
        </Form>
      );

      expect(getByText('ojs_togglebutton_on')).toBeVisible();
      expect(getByText('ojs_togglebutton_on')).toBeDisabled();

      expect(getByText('ojs_togglebutton_off')).toBeVisible();
      expect(getByText('ojs_togglebutton_off')).toBeDisabled();
    });
  });

  const testButton = (initialValue: boolean): void => {
    describe(`Button with initial value: ${initialValue}`, () => {
      it('should set the correct button as active', () => {
        const { getByText } = render(
          <Form>
            <OnOffToggleButton name="mock" label="mock" value={initialValue} onLabel="on" offLabel="off" />
          </Form>
        );

        const activeButton = initialValue ? 'on' : 'off';
        const inactiveButton = initialValue ? 'off' : 'on';

        expect(getByText(activeButton)).toHaveClass('active');
        expect(getByText(inactiveButton)).not.toHaveClass('active');
      });

      it('should trigger onChange on click', () => {
        const handleChange = jest.fn();
        const { getByText } = render(
          <Form>
            <OnOffToggleButton name="mock" label="mock" value={initialValue} onLabel="on" offLabel="off" onChange={handleChange} />
          </Form>
        );

        const inactiveButton = initialValue ? 'off' : 'on';
        fireEvent.click(getByText(inactiveButton));

        expect(handleChange).toHaveBeenCalledTimes(1);
        expect(handleChange).toHaveBeenCalledWith(!initialValue);

        expect(getByText(inactiveButton)).toHaveClass('active');
      });
    });
  };

  testButton(true);
  testButton(false);
});
