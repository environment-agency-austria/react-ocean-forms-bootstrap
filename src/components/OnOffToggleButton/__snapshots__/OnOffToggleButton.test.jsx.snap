// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseOnOffToggleButton /> should render correctly in disabled mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OnOffToggleButton
    field={
      Object {
        "disabled": true,
        "id": "mock-field",
        "name": "mock-field",
        "onBlur": [MockFunction],
        "onChange": [MockFunction],
        "value": false,
      }
    }
    label="field0"
    meta={
      Object {
        "error": null,
        "isValidating": false,
        "plaintext": true,
        "stringFormatter": [MockFunction],
        "touched": false,
        "valid": true,
      }
    }
    offLabel="off"
    onLabel="on"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "field": Object {
        "disabled": true,
        "id": "mock-field",
        "name": "mock-field",
        "onBlur": [MockFunction],
        "onChange": [MockFunction],
        "value": false,
      },
      "label": "field0",
      "meta": Object {
        "error": null,
        "isValidating": false,
        "plaintext": true,
        "stringFormatter": [MockFunction],
        "touched": false,
        "valid": true,
      },
      "offLabel": "off",
      "onLabel": "on",
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "field": Object {
          "disabled": true,
          "id": "mock-field",
          "name": "mock-field",
          "onBlur": [MockFunction],
          "onChange": [MockFunction],
          "value": false,
        },
        "label": "field0",
        "meta": Object {
          "error": null,
          "isValidating": false,
          "plaintext": true,
          "stringFormatter": [MockFunction],
          "touched": false,
          "valid": true,
        },
        "offLabel": "off",
        "onLabel": "on",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<BaseOnOffToggleButton /> should render plaintext Button 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OnOffToggleButton
    field={
      Object {
        "disabled": false,
        "id": "mock-field",
        "name": "mock-field",
        "onBlur": [MockFunction],
        "onChange": [MockFunction],
        "value": "mock-value",
      }
    }
    label="field0"
    meta={
      Object {
        "error": null,
        "isValidating": false,
        "plaintext": true,
        "stringFormatter": [MockFunction],
        "touched": false,
        "valid": true,
      }
    }
    offLabel="off"
    onLabel="on"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "field": Object {
        "disabled": false,
        "id": "mock-field",
        "name": "mock-field",
        "onBlur": [MockFunction],
        "onChange": [MockFunction],
        "value": "mock-value",
      },
      "label": "field0",
      "meta": Object {
        "error": null,
        "isValidating": false,
        "plaintext": true,
        "stringFormatter": [MockFunction],
        "touched": false,
        "valid": true,
      },
      "offLabel": "off",
      "onLabel": "on",
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "field": Object {
          "disabled": false,
          "id": "mock-field",
          "name": "mock-field",
          "onBlur": [MockFunction],
          "onChange": [MockFunction],
          "value": "mock-value",
        },
        "label": "field0",
        "meta": Object {
          "error": null,
          "isValidating": false,
          "plaintext": true,
          "stringFormatter": [MockFunction],
          "touched": false,
          "valid": true,
        },
        "offLabel": "off",
        "onLabel": "on",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<BaseOnOffToggleButton /> should render the correct label in plaintext mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OnOffToggleButton
    field={
      Object {
        "disabled": false,
        "id": "mock-field",
        "name": "mock-field",
        "onBlur": [MockFunction],
        "onChange": [MockFunction],
        "value": false,
      }
    }
    label="field0"
    meta={
      Object {
        "error": null,
        "isValidating": false,
        "plaintext": true,
        "stringFormatter": [MockFunction],
        "touched": false,
        "valid": true,
      }
    }
    offLabel="off"
    onLabel="on"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "field": Object {
        "disabled": false,
        "id": "mock-field",
        "name": "mock-field",
        "onBlur": [MockFunction],
        "onChange": [MockFunction],
        "value": false,
      },
      "label": "field0",
      "meta": Object {
        "error": null,
        "isValidating": false,
        "plaintext": true,
        "stringFormatter": [MockFunction],
        "touched": false,
        "valid": true,
      },
      "offLabel": "off",
      "onLabel": "on",
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "field": Object {
          "disabled": false,
          "id": "mock-field",
          "name": "mock-field",
          "onBlur": [MockFunction],
          "onChange": [MockFunction],
          "value": false,
        },
        "label": "field0",
        "meta": Object {
          "error": null,
          "isValidating": false,
          "plaintext": true,
          "stringFormatter": [MockFunction],
          "touched": false,
          "valid": true,
        },
        "offLabel": "off",
        "onLabel": "on",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<BaseOnOffToggleButton /> should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OnOffToggleButton
    field={
      Object {
        "disabled": false,
        "id": "mock-field",
        "name": "mock-field",
        "onBlur": [MockFunction],
        "onChange": [MockFunction],
        "value": "mock-value",
      }
    }
    label="field0"
    meta={
      Object {
        "error": null,
        "isValidating": false,
        "plaintext": false,
        "stringFormatter": [MockFunction],
        "touched": false,
        "valid": true,
      }
    }
    offLabel="off"
    onLabel="on"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ButtonGroup
        id="mock-field"
        role="group"
        tag="div"
      >
        <Button
          active={true}
          color="primary"
          disabled={false}
          id="mock-field-on"
          onClick={[Function]}
          outline={true}
          tag="button"
        >
          <FormComponent(FormText)
            text="on"
          />
        </Button>
        <Button
          active={false}
          color="primary"
          disabled={false}
          id="mock-field-off"
          onClick={[Function]}
          outline={true}
          tag="button"
        >
          <FormComponent(FormText)
            text="off"
          />
        </Button>
      </ButtonGroup>,
      "field": Object {
        "disabled": false,
        "id": "mock-field",
        "name": "mock-field",
        "onBlur": [MockFunction],
        "onChange": [MockFunction],
        "value": "mock-value",
      },
      "inputSize": "9",
      "label": "field0",
      "labelClass": "text-right",
      "labelSize": "3",
      "meta": Object {
        "error": null,
        "isValidating": false,
        "plaintext": false,
        "stringFormatter": [MockFunction],
        "touched": false,
        "valid": true,
      },
      "offLabel": "off",
      "onLabel": "on",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Button
            active={true}
            color="primary"
            disabled={false}
            id="mock-field-on"
            onClick={[Function]}
            outline={true}
            tag="button"
          >
            <FormComponent(FormText)
              text="on"
            />
          </Button>,
          <Button
            active={false}
            color="primary"
            disabled={false}
            id="mock-field-off"
            onClick={[Function]}
            outline={true}
            tag="button"
          >
            <FormComponent(FormText)
              text="off"
            />
          </Button>,
        ],
        "id": "mock-field",
        "role": "group",
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": true,
            "children": <FormComponent(FormText)
              text="on"
            />,
            "color": "primary",
            "disabled": false,
            "id": "mock-field-on",
            "onClick": [Function],
            "outline": true,
            "tag": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "text": "on",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "children": <FormComponent(FormText)
              text="off"
            />,
            "color": "primary",
            "disabled": false,
            "id": "mock-field-off",
            "onClick": [Function],
            "outline": true,
            "tag": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "text": "off",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ButtonGroup
          id="mock-field"
          role="group"
          tag="div"
        >
          <Button
            active={true}
            color="primary"
            disabled={false}
            id="mock-field-on"
            onClick={[Function]}
            outline={true}
            tag="button"
          >
            <FormComponent(FormText)
              text="on"
            />
          </Button>
          <Button
            active={false}
            color="primary"
            disabled={false}
            id="mock-field-off"
            onClick={[Function]}
            outline={true}
            tag="button"
          >
            <FormComponent(FormText)
              text="off"
            />
          </Button>
        </ButtonGroup>,
        "field": Object {
          "disabled": false,
          "id": "mock-field",
          "name": "mock-field",
          "onBlur": [MockFunction],
          "onChange": [MockFunction],
          "value": "mock-value",
        },
        "inputSize": "9",
        "label": "field0",
        "labelClass": "text-right",
        "labelSize": "3",
        "meta": Object {
          "error": null,
          "isValidating": false,
          "plaintext": false,
          "stringFormatter": [MockFunction],
          "touched": false,
          "valid": true,
        },
        "offLabel": "off",
        "onLabel": "on",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Button
              active={true}
              color="primary"
              disabled={false}
              id="mock-field-on"
              onClick={[Function]}
              outline={true}
              tag="button"
            >
              <FormComponent(FormText)
                text="on"
              />
            </Button>,
            <Button
              active={false}
              color="primary"
              disabled={false}
              id="mock-field-off"
              onClick={[Function]}
              outline={true}
              tag="button"
            >
              <FormComponent(FormText)
                text="off"
              />
            </Button>,
          ],
          "id": "mock-field",
          "role": "group",
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": true,
              "children": <FormComponent(FormText)
                text="on"
              />,
              "color": "primary",
              "disabled": false,
              "id": "mock-field-on",
              "onClick": [Function],
              "outline": true,
              "tag": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "text": "on",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "children": <FormComponent(FormText)
                text="off"
              />,
              "color": "primary",
              "disabled": false,
              "id": "mock-field-off",
              "onClick": [Function],
              "outline": true,
              "tag": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "text": "off",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
